src/extra/recipes/cookiebox.py

```diff
@@ -0,0 +1,1051 @@
+        ax[2].plot(self.energy_axis, self.offset[tof_id], lw=2, label="Interpolated offset")
+        ax[2].set(xlabel="Energy [eV]", ylabel="Offset to subtract [a.u.]")
+        ax[3].scatter(self.tof_fit_result[tof_id].energy, self.tof_fit_result[tof_id].Aa/self.calibration_mean_xgm[tof_id], label="Normalization")
+        ax[3].plot(self.energy_axis, self.normalization[tof_id], lw=2, label="Interpolated normalization")
+        ax[3].set(xlabel="Energy [eV]", ylabel="(Auger+valence)/pulse energy [a.u.]")
+        ax[4].plot(self.energy_axis, self.jacobian[tof_id], lw=2, label="Interpolated Jacobian")
+        ax[4].set(xlabel="Energy [eV]", ylabel="Jacobian [a.u.]")
+        for a in ax:
+            a.set_title(f"TOF {tof_id}")
+        plt.show()
+
+    def plot_all_transmissions(self, tof_ids: Optional[List[int]]=None):
+        """
+        Plot all transmissions in the same plot.
+
+        Args:
+          tof_ids: List of eTOF IDs to plot. If none, show all.
+        """
+        import matplotlib.pyplot as plt
+        fig = plt.figure(figsize=(20, 10))
+        for tof_id, transmission in self.int_transmission.items():
+            if tof_ids is not None:
+                if tof_id not in tof_ids:
+                    continue
+            plt.plot(self.energy_axis, transmission, lw=2, label=f"eTOF {tof_id}")
+        plt.xlabel("Energy [eV]")
+        plt.ylabel("Transmission [a.u.]")
+        plt.legend()
+        plt.show()
+
+    def plot_all_offsets(self, tof_ids: Optional[List[int]]=None):
+        """
+        Plot all offset in the same plot.
+
+        Args:
+          tof_ids: List of eTOF IDs to plot. If none, show all.
+        """
+        import matplotlib.pyplot as plt
+        fig = plt.figure(figsize=(20, 10))
+        for tof_id, offset in self.offset.items():
+            if tof_ids is not None:
+                if tof_id not in tof_ids:
+                    continue
+            plt.plot(self.energy_axis, offset, lw=2, label=f"eTOF {tof_id}")
+        plt.xlabel("Energy [eV]")
+        plt.ylabel("Offset [a.u.]")
+        plt.legend()
+        plt.show()
+
+    def plot_all_jacobians(self, tof_ids: Optional[List[int]]=None):
+        """
+        Plot all offset in the same plot.
+
+        Args:
+          tof_ids: List of eTOF IDs to plot. If none, show all.
+        """
+        import matplotlib.pyplot as plt
+        fig = plt.figure(figsize=(20, 10))
+        for tof_id, jacobian in self.jacobian.items():
+            if tof_ids is not None:
+                if tof_id not in tof_ids:
+                    continue
+            plt.plot(self.energy_axis, jacobian, lw=2, label=f"eTOF {tof_id}")
+        plt.xlabel("Energy [eV]")
+        plt.ylabel("Jacobian [1/eV]")
+        plt.legend()
+        plt.show()
+
+    def plot_all_normalizations(self, tof_ids: Optional[List[int]]=None):
+        """
+        Plot all normalizations in the same plot.
```

Comment:
Minor thing, I would recommend making screenshots of the plots these generates and linking them in the docstring to give the users an idea of what to expect.

Reply:
I might do this later. I am not sure if these plots are ok to share (it is not commissioning data). It is nothing of consequence, but it might be worth checking.